### PR TITLE
Fix #15790

### LINKED ISSUE #15790: Increase the visibility of InputText._selectAllTexts() public, or provide an public alternative
**Is your feature request related to a problem? Please describe.**
Currently, there's no way to select the content of an `InputText` control programmatically. There is `_selectAllText()`, but it's a protected method, so it's unsuitable for ordinary use cases.

**Describe the solution you'd like**
I want either the `_selectAllText()` method to be a public method or an alternative public API to be provided.

**Discussion**
You must provide a link to the forum post where the initial discussion happened:

https://forum.babylonjs.com/t/is-there-an-alternative-to-inputtext-selectalltext/54636

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15797/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15797/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15797/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15797/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15797/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15797/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15797/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15797/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15797/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15797/merge/testResults/webgpuplaywright/index.html

File: packages/dev/gui/src/2D/controls/inputText.ts [hunk 1]
```diff
@@ -812,6 +812,13 @@ export class InputText extends Control {
         this._markAsDirty();
     }
 
+    /**
+     * Allow the user to select all text
+     */
+    public selectAllText() {
```
Comment (sebavan) [inline packages/dev/gui/src/2D/controls/inputText.ts:818]: Any reason not to make _selectAllText public ? Wondering if inputTextArea would not benefit from it as well ?

Comment (deltakosh) [reply packages/dev/gui/src/2D/controls/inputText.ts:818]: good point! I will

**Review by sebavan — APPROVED** [review]

**Pull request merged**